`gh pr list` command is broken with install from source

### Describe the bug

```sh
❯ gh --version
gh version 2.35.0-13-g5dc58ca6 (2023-09-27)
https://github.com/cli/cli/releases/latest
```

### Steps to reproduce the behavior

```sh
git clone https://github.com/cli/cli.git gh-cli
cd gh-cli
sudo make install
```

(just `make install` command become access error about `/usr/local/bin/` so I add `sudo`)

then


```sh
❯ gh --version
gh version 2.35.0-13-g5dc58ca6 (2023-09-27)
https://github.com/cli/cli/releases/latest
```

but, gh pr command become panic

```sh
gh pr list
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x12b1b64]

goroutine 1 [running]:
github.com/cli/cli/v2/pkg/cmd/root.NewCmdRoot.func1(0xc00063e600, {0x452569?, 0x417578?, 0x14eb088?})
	github.com/cli/cli/v2/pkg/cmd/root/root.go:75 +0x64
github.com/cli/cli/v2/pkg/cmdutil.executeParentHooks(0xe500000000ea36f8?, {0x2d82860?, 0x2d82860?, 0x2?})
	github.com/cli/cli/v2/pkg/cmdutil/repo_override.go:16 +0x30
github.com/cli/cli/v2/pkg/cmdutil.EnableRepoOverride.func2(0x14ee27a?, {0x2d82860?, 0x14eb088?, 0x4?})
	github.com/cli/cli/v2/pkg/cmdutil/repo_override.go:51 +0x2a
github.com/cli/cli/v2/pkg/cmdutil.executeParentHooks(0xc000170820?, {0x2d82860?, 0x0?, 0x0?})
	github.com/cli/cli/v2/pkg/cmdutil/repo_override.go:16 +0x30
github.com/cli/cli/v2/pkg/cmdutil.EnableRepoOverride.func2(0xc000574b00?, {0x2d82860?, 0x4?, 0x14eb084?})
	github.com/cli/cli/v2/pkg/cmdutil/repo_override.go:51 +0x2a
github.com/spf13/cobra.(*Command).execute(0xc000573800, {0x2d82860, 0x0, 0x0})
	github.com/spf13/cobra@v1.6.1/command.go:891 +0x75f
github.com/spf13/cobra.(*Command).ExecuteC(0xc00063e600)
	github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3a5
github.com/spf13/cobra.(*Command).ExecuteContextC(...)
	github.com/spf13/cobra@v1.6.1/command.go:977
main.mainRun()
	github.com/cli/cli/v2/cmd/gh/main.go:110 +0x513
main.main()
	github.com/cli/cli/v2/cmd/gh/main.go:45 +0x13
```


Additional Info 

```
OS: Manjaro Linux x86_64
Kernel: 6.2.16-1-MANJARO
```

--- a/pkg/cmd/root/root.go
+++ b/pkg/cmd/root/root.go
@@ -72,7 +72,8 @@ func NewCmdRoot(f *cmdutil.Factory, version, buildDate string) (*cobra.Command, 
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
 			// require that the user is authenticated before running most commands
 			if cmdutil.IsAuthCheckEnabled(cmd) && !cmdutil.CheckAuth(cfg) {
-				if cmd.Parent().Use == "codespace" {
+				parent := cmd.Parent()
+				if parent != nil && parent.Use == "codespace" {
 					fmt.Fprintln(io.ErrOut, "To get started with GitHub CLI, please run:  gh auth login -s codespace")
 				} else {
 					fmt.Fprint(io.ErrOut, authHelp())

